x-pack/plugins/security_solution/common/endpoint/data_generators/base_data_generator.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import seedrandom from 'seedrandom';
+import uuid from 'uuid';
+
+const OS_FAMILY = ['windows', 'macos', 'linux'];
+
+/**
+ * A generic base class to assist in creating domain specific data generators. It includes
+ * several general purpose random data generators for use within the class and exposes one
+ * public method named `generate()` which should be implemented by sub-classes.
+ */
+export class BaseDataGenerator<GeneratedDoc extends {} = {}> {
```

Comment:
Crated a base class for use in domain specific generators, that includes only generic type of random data producers